docs/serving/configuration/config-defaults.md

```diff
@@ -30,319 +32,497 @@ data:
-{{.Name}} are also valid.
+The `container-name-template` value provides a template for the default
+container name if no container name is specified. This field supports Go templating and is supplied by the `ObjectMeta` of the enclosing Service or Configuration, so values such as `{{.Name}}` are also valid.
 {% endraw %}
 
-**Key**: `container-name-template`
-
-**Default**: `"user-container"`
+* **Global key:** `container-name-template`
+* **Per-revision annotation key:** `serving.knative.dev/container-name-template`
+* **Possible values:** string
+* **Default:** `"user-container"`
 
 **Example:**
-```yaml
-apiVersion:  v1
-kind:  ConfigMap
-metadata:
-  name:  config-defaults
-  namespace:  knative-serving
-data:
-  container-name-template: "user-container"
-```
 
-### Container Concurrency
+=== "Global (ConfigMap)"
+    ```yaml
+    apiVersion:  v1
+    kind:  ConfigMap
+    metadata:
+      name:  config-defaults
+      namespace:  knative-serving
+    data:
+      container-name-template: "user-container"
+    ```
+
+=== "Per Revision"
+    ```yaml
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    metadata:
+      name: helloworld-go
+      namespace: default
+    spec:
+      template:
+        metadata:
+          annotations:
+            serving.knative.dev/container-name-template: "user-container"
+        spec:
+          containers:
+            - image: gcr.io/knative-samples/helloworld-go
+    ```
+
+### Container concurrency
 {% raw %}
-container-concurrency specifies the maximum number
-of requests the Container can handle at once, and requests
-above this threshold are queued.  Setting a value of zero
-disables this throttling and lets through as many requests as
+The `container-concurrency` value specifies the maximum number of requests the container can handle at once. Requests above this threshold are queued. Setting a value of zero disables this throttling and lets through as many requests as
 the pod receives.
 {% endraw %}
 
-**Key**: `container-concurrency`
-
-**Default**: `"0"`
+* **Global key:** `container-concurrency`
+* **Per-revision annotation key:** `serving.knative.dev/container-concurrency`
+* **Possible values:** integer
+* **Default:** `"0"`
 
 **Example:**
-```yaml
-apiVersion:  v1
-kind:  ConfigMap
-metadata:
-  name:  config-defaults
-  namespace:  knative-serving
-data:
-  container-concurrency: "0"
-```
 
-### Container Concurrency Max Limit
+=== "Global (ConfigMap)"
+    ```yaml
+    apiVersion:  v1
+    kind:  ConfigMap
+    metadata:
+      name:  config-defaults
+      namespace:  knative-serving
+    data:
+      container-concurrency: "0"
+    ```
+
+=== "Per Revision"
+    ```yaml
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    metadata:
+      name: helloworld-go
+      namespace: default
+    spec:
+      template:
+        metadata:
+          annotations:
+            serving.knative.dev/container-name-template: "0"
```

Comment:
typo in the annotation - but it doesn't exist

The container-concurrency annotation maps to `Spec.ContainerConcurrency` on the `Revision`